OpenRA.Mods.Common/Warheads/SpreadDamageWarhead.cs

```diff
@@ -66,11 +67,17 @@ public override void DoImpact(WPos pos, Actor firedBy, IEnumerable<int> damageMo
 
 			foreach (var victim in hitActors)
 			{
+				// Cannot be damaged without a Health trait
 				var healthInfo = victim.Info.TraitInfoOrDefault<HealthInfo>();
 				if (healthInfo == null)
 					continue;
 
-				var distance = healthInfo.Shape.DistanceFromEdge(pos, victim);
+				// Cannot be damaged without an active HitShape
+				var activeShapes = victim.TraitsImplementing<HitShape>().Where(Exts.IsTraitEnabled).ToArray();
```

Comment:
For transient collections that you only enumerate once `ToArray` actually hurts performance (you need to enumerate it twice to create the array, and then a third time to do the thing you want to do).